ballista/executor/src/flight_service.rs

```diff
@@ -68,6 +71,54 @@ impl BallistaFlightService {
 type BoxedFlightStream<T> =
     Pin<Box<dyn Stream<Item = Result<T, Status>> + Send + 'static>>;
 
+fn decode_partition_location(ticket: &Ticket) -> Result<String, Status> {
+    match &decode_protobuf(&ticket.ticket).map_err(|e| from_ballista_err(&e))? {
+        BallistaAction::FetchPartition { path, .. } => Ok(path.clone()),
+    }
+}
+
+async fn do_get_aux(
+    executor_id: String,
+    path: &str,
+) -> Result<Response<BoxedFlightStream<FlightData>>, Status> {
+    info!(
+        executor_id = executor_id,
+        path, "fetching shuffle partition"
+    );
+    let file = File::open(path)
+        .map_err(|e| {
+            BallistaError::General(format!(
+                "Failed to open partition file at {path}: {e:?}"
+            ))
+        })
+        .map_err(|e| from_ballista_err(&e))?;
+    let reader = FileReader::try_new(file, None).map_err(|e| from_arrow_err(&e))?;
```

Comment:
In theory we could just stream the bytes and return them in the stream (because arrow files is IPC as well)

Reply:
Done